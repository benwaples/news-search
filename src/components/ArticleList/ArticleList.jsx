import React from 'react';
import PropTypes from 'prop-types';
import styles from './ArticleList.css';
import Article from '../Article/Article';

function ArticleList({ articles }) {
  const ArticleElements = articles.map(article => (
    <Article key={article.title} article={article}/>
  ));

  return (
    <div id={styles.articleList}>
      <ul>
        {ArticleElements}
      </ul>
    </div>
  );
}

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    sourceName: PropTypes.string,
    title: PropTypes.string,
    author: PropTypes.string,
    content: PropTypes.string,
    url: PropTypes.string
  })).isRequired
};

export default ArticleList;

