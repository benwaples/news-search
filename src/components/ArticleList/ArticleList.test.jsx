import React from 'react';
import { 
  cleanup, 
  fireEvent, 
  render, 
  screen, 
  waitFor 
} from '@testing-library/react';
import ArticleList from './ArticleList';

describe('test the ArticleList component', () => {
  afterEach(() => cleanup());

  it('should render a list of articles', () => {
    const articles = [
      {
        source: 'article source',
        title: 'test title3',
        author: 'fake af author',
        content: 'some true stuff',
        imageUrl: 'https://fakeplaceholder.image.url.you/couldnt',
        url: 'google.com',
      },
      {
        source: 'article source',
        title: 'test title2',
        author: 'fake af author',
        content: 'some true stuff',
        imageUrl: 'https://fakeplaceholder.image.url.you/couldnt',
        url: 'google.com',
      },
      {
        source: 'article source',
        title: 'test title1',
        author: 'fake af author',
        content: 'some true stuff',
        imageUrl: 'https://fakeplaceholder.image.url.you/couldnt',
        url: 'google.com',
      }
    ];

    const { asFragment } = render(<ArticleList articles={articles} />);

    const allArticleTitles = screen.getAllByText('some true stuff');

    expect(allArticleTitles.length).toEqual(3);

    expect(asFragment()).toMatchSnapshot();
  });
});
