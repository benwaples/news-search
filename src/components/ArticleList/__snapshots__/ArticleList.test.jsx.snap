// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the ArticleList component should render a list of articles 1`] = `
<DocumentFragment>
  <div
    id="articleList"
  >
    <ul>
      <li
        class="Article"
      >
        <div>
          <h1
            data-testid="article"
          >
            test title3
          </h1>
          <p>
            article source
          </p>
          <p>
            fake af author
          </p>
        </div>
        <p>
          some true stuff
        </p>
        <a
          href="google.com"
        >
          <img
            alt="test title3"
            src="https://fakeplaceholder.image.url.you/couldnt"
          />
        </a>
        <p>
          click above to see more
        </p>
      </li>
      <li
        class="Article"
      >
        <div>
          <h1
            data-testid="article"
          >
            test title2
          </h1>
          <p>
            article source
          </p>
          <p>
            fake af author
          </p>
        </div>
        <p>
          some true stuff
        </p>
        <a
          href="google.com"
        >
          <img
            alt="test title2"
            src="https://fakeplaceholder.image.url.you/couldnt"
          />
        </a>
        <p>
          click above to see more
        </p>
      </li>
      <li
        class="Article"
      >
        <div>
          <h1
            data-testid="article"
          >
            test title1
          </h1>
          <p>
            article source
          </p>
          <p>
            fake af author
          </p>
        </div>
        <p>
          some true stuff
        </p>
        <a
          href="google.com"
        >
          <img
            alt="test title1"
            src="https://fakeplaceholder.image.url.you/couldnt"
          />
        </a>
        <p>
          click above to see more
        </p>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
